Fixed #4947: Buddy palette appears in wrong position

AbrahmAB opened the pull request:
Fixed [#4947](https://bugs.sugarlabs.org/ticket/4947)  . The palette was created when the cursor hovered in the buddy column even though there was no buddy icon at cursor position!

quozl commented:
Reproduced reported problem on Fedora 18 and Ubuntu 15.10 with Sugar 0.108.0.

Tested on Fedora 18 and Ubuntu 15.10; your patch does fix the problem.

Reviewed patch;
* the commit message doesn't explain the problem or the solution,
* the commented assignment of *column_index* is puzzling, as it isn't used or explained,
* the use of an index variable *i* is unfortunate, but is used twice inside the for loop, makes me wonder if there is an alternative solution that changes data structures to make it easier.

AbrahmAB commented:
@quozl 
Changes made in commit
- Amended the commit message.

- Removed unnecessary comment

- Implemented the loop in better way by replacing the variable _i_ with _column_index_

quozl commented:
Reviewed, thanks.
